homeassistant/components/freedompro/light.py

```diff
@@ -0,0 +1,131 @@
+"""Support for Freedompro light."""
+import json
+import math
+
+from pyfreedompro import put_state
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    ATTR_HS_COLOR,
+    COLOR_MODE_BRIGHTNESS,
+    COLOR_MODE_HS,
+    LightEntity,
+)
+from homeassistant.const import CONF_API_KEY
+from homeassistant.core import callback
+from homeassistant.helpers import aiohttp_client
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up Freedompro light."""
+    api_key = entry.data[CONF_API_KEY]
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        Device(hass, api_key, device, coordinator)
+        for device in coordinator.data
+        if device["type"] == "lightbulb"
+    )
+
+
+class Device(CoordinatorEntity, LightEntity):
+    """Representation of an Freedompro light."""
+
+    def __init__(self, hass, api_key, device, coordinator):
+        """Initialize the Freedompro light."""
+        super().__init__(coordinator)
+        self._hass = hass
+        self._session = aiohttp_client.async_get_clientsession(self._hass)
+        self._api_key = api_key
+        self._attr_name = device["name"]
+        self._attr_unique_id = device["uid"]
+        self._type = device["type"]
+        self._characteristics = device["characteristics"]
+        self._on = False
+        self._brightness = 0
+        self._saturation = 0
+        self._hue = 0
+
+    @property
+    def supported_features(self):
```

Comment:
A device can only be in a single color mode at the time.
For color modes you need to provide the `color_mode` and `color_modes` properties.
See: <https://developers.home-assistant.io/docs/core/entity/light>